attestation-service/verifier/src/se/ibmse.rs

```diff
@@ -0,0 +1,311 @@
+// Copyright (C) Copyright IBM Corp. 2024
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use crate::TeeEvidenceParsedClaim;
+use anyhow::{anyhow, bail, Context, Result};
+use core::result::Result::Ok;
+use log::{debug, info, warn};
+use openssl::encrypt::{Decrypter, Encrypter};
+use openssl::pkey::{PKey, Private, Public};
+use openssl::rsa::{Padding, Rsa};
+use pv::attest::{
+    AdditionalData, AttestationFlags, AttestationItems, AttestationMeasAlg, AttestationMeasurement,
+    AttestationRequest, AttestationVersion,
+};
+use pv::misc::{open_file, read_certs};
+use pv::request::{BootHdrTags, CertVerifier, HkdVerifier, ReqEncrCtx, Request, SymKeyType};
+use pv::uv::ConfigUid;
+use serde::{Deserialize, Serialize};
+use serde_with::{base64::Base64, serde_as};
+use std::{env, fs};
+
+const DEFAULT_SE_HOST_KEY_DOCUMENTS_ROOT: &str = "/run/confidential-containers/ibmse/hkds";
+
+const DEFAULT_SE_CERTIFICATES_ROOT: &str = "/run/confidential-containers/ibmse/certs";
+
+const DEFAULT_SE_CERTIFICATE_ROOT_CA: &str = "/run/confidential-containers/ibmse/certs/ca";
+
+const DEFAULT_SE_CERTIFICATE_REVOCATION_LISTS_ROOT: &str =
+    "/run/confidential-containers/ibmse/crls";
+
+const DEFAULT_SE_IMAGE_HEADER_FILE: &str = "/run/confidential-containers/ibmse/hdr/hdr.bin";
+
+const DEFAULT_SE_MEASUREMENT_ENCR_KEY_PRIVATE: &str =
+    "/run/confidential-containers/ibmse/rsa/encrypt_key.pem";
+
+const DEFAULT_SE_MEASUREMENT_ENCR_KEY_PUBLIC: &str =
+    "/run/confidential-containers/ibmse/rsa/encrypt_key.pub";
+
+macro_rules! env_or_default {
+    ($env:literal, $default:ident) => {
+        match env::var($env) {
+            Ok(env_path) => env_path,
+            Err(_) => $default.into(),
+        }
+    };
+}
+
+fn list_files_in_folder(dir: &str) -> Result<Vec<String>> {
+    let mut file_paths = Vec::new();
+
+    for entry in fs::read_dir(dir)? {
+        let entry = entry?;
+        let path = entry.path();
+
+        if path.is_file() {
+            if let Some(path_str) = path.to_str() {
+                file_paths.push(path_str.to_string());
+            }
+        }
+    }
+
+    Ok(file_paths)
+}
+
+#[serde_as]
+#[derive(Debug, Serialize, Deserialize)]
+pub struct SeAttestationResponse {
+    #[serde_as(as = "Base64")]
+    measurement: Vec<u8>,
+    #[serde_as(as = "Base64")]
+    additional_data: Vec<u8>,
+    #[serde_as(as = "Base64")]
+    user_data: Vec<u8>,
+    #[serde_as(as = "Base64")]
+    cuid: ConfigUid,
+    #[serde_as(as = "Base64")]
+    encr_measurement_key: Vec<u8>,
+    #[serde_as(as = "Base64")]
+    encr_request_nonce: Vec<u8>,
+    #[serde_as(as = "Base64")]
+    image_hdr_tags: BootHdrTags,
+}
+
+#[repr(C)]
+#[serde_as]
+#[derive(Debug, Serialize, Deserialize)]
+pub struct SeAttestationClaims {
+    #[serde_as(as = "Base64")]
```

Comment:
I suggest that we use hex/base16 to encode the parsed claims. This would help users have a common way to set policy. see https://github.com/confidential-containers/trustee/issues/372

Reply:
Yeah, I'm happy to keep aligned with other drivers.